src/openshift/cluster.ts

```diff
@@ -478,6 +534,61 @@ export class Cluster extends OpenShiftItem {
         }
     }
 
+    private static ongoingOperationCanceller: AbortController;
+
+    /**
+     * Checks if a new operation can be started.
+     * If an ongoing operation is already started, asks if user really wants to cancel it and start a new one,
+     * If user agrees - the ongoing operation gets canceled.
+     *
+     * @returns true if we can start a new operation, false - otherwise
+     */
+    private static async checkOngoingOperation(abortController: AbortController): Promise<boolean> {
+        if (Cluster.ongoingOperationCanceller && Cluster.ongoingOperationCanceller !== abortController) {
+            let response = 'Yes';
+            const cluster = new KubeConfigUtils().getCurrentCluster();
+            response = await window.showInformationMessage(
+                `You are already trying tp login to ${cluster ? cluster.server : ''} cluster. Do you want to login to cancel and try again?`,
+                'Yes',
+                'No',
+            );
+            if (response !== 'Yes') {
+                return false;
+            }
+
+            // Do cancell here.
+            Cluster.ongoingOperationCanceller && Cluster.ongoingOperationCanceller.abort();
+        }
+
+        return true;
+    }
+
+    /**
+     * Checks the availability of a given cluster
+     *
+     * @param url Cluster URL to check
+     * @param abortController if provided, allows cancelling the operation
+     * @returns true if cluster is available, false if cluster is not available or the operation is cancelled
+     */
+    static async pingCluster(url: string, abortController: AbortController): Promise<boolean> {
+        return new Promise<boolean>((resolve, reject) => {
+            const signal = abortController?.signal;
+            const options = { rejectUnauthorized: false, signal };
+            const https = require('https');
+            https.get(`${url}/api`, options, (response) => {
```

Comment:
In this iteration it's changed to `import * as https from 'https'`

Reply:
Regarding the use of node's `fetch`. The common way to use it looks like:

```
import * as https from 'https';
import fetch from 'node-fetch'

const signal = abortController?.signal;
fetch(`${url}/api`, { agent: new https.Agent({rejectUnauthorized: false}), signal })
	.then((response) => {
	    if (response.status < 500) {
		resolve(true);
	    } else {
		reject(`Connect error: ${response.statusText}`);
	    }
	})
	.catch((reason) => {
	    reject(`Connect error: ${reason}`);
	});
```

... And the problem here is that use of `https.Agent`'s option `rejectUnauthorized: false` (to skip SSL certificate validation ) has no effect on Node.js v.18 and v.20 (at least according to comments on StackOverflow, for example, see the comments under this post: https://stackoverflow.com/a/59944400/24035061). 

So, the requests to get `${clusterURL}/api` get rejected with the following error:
```
FetchError {
  message: 'request to https://api.ocp2.adapters-crs.cci…self signed certificate in certificate chain', 
  type: 'system', 
  errno: 'SELF_SIGNED_CERT_IN_CHAIN', 
  code: 'SELF_SIGNED_CERT_IN_CHAIN', 
  stack: 'FetchError: request to https://api.ocp2.adapt…ons (node:internal/process/task_queues:82:21)'
}
```

So, it looks like using Node's `fetch` doesn't make sense at the moment.

Reply:
node-fetch predates fetch being built-in to node, so it might have a different API

Reply:
> You shouldn't need to import anything to use node's fetch, since it is built in
...
> node-fetch predates fetch being built-in to node, so it might have a different API

That was an outdated code: currently I don't import neither `make-fetch-happen` nor `node-fetch`. 
The only thing I need is `https` which allows to create and configure `Agent` trying to turn on the SSL Certificate verification (which doesn't work though).

Reply:
You're right, there is no way to avoid rejecting invalid certificates using node's built-in fetch (unless you disable it for all https requests which is really bad)